op-program/client/tasks/derive.go

```diff
@@ -54,6 +59,10 @@ func RunDerivation(
 		return DerivationResult{}, fmt.Errorf("failed to run program to completion: %w", err)
 	}
 	logger.Info("Derivation complete", "head", result)
+	if err := writeTrieNodes(result, db, engineBackend); err != nil {
+		return DerivationResult{}, fmt.Errorf("failed to write trie nodes: %w", err)
+	}
+	logger.Info("Trie nodes written")
```

Comment:
I wonder if we should make this an optional step via something in the client config. Then we could skip it during normal derivation and only do it when regenerating a block.

Reply:
I considered doing this but wasn't sure if it's worth it to have more conditional logic in the program. But maybe it's not too bad. Lemme whip up something.